src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+        {
+            List<CodeAction> codeActions = new List<CodeAction>();
+
+            await codeFixProvider.RegisterCodeFixesAsync(new CodeFixContext(solution.GetDocument(diagnostic.Location.SourceTree), diagnostic, (a, d) => codeActions.Add(a), cancellationToken)).ConfigureAwait(false);
+
+            return codeActions;
+        }
+
+        private static Task<Statistic> GetAnalyzerStatisticsAsync(IEnumerable<Project> projects, CancellationToken cancellationToken)
+        {
+            ConcurrentBag<Statistic> sums = new ConcurrentBag<Statistic>();
+
+            Parallel.ForEach(projects.SelectMany(i => i.Documents), document =>
+            {
+                var documentStatistics = GetAnalyzerStatisticsAsync(document, cancellationToken).ConfigureAwait(false).GetAwaiter().GetResult();
+                sums.Add(documentStatistics);
+            });
+
+            Statistic sum = sums.Aggregate(new Statistic(0, 0, 0), (currentResult, value) => currentResult + value);
+            return Task.FromResult(sum);
+        }
+
+        private static async Task<Statistic> GetAnalyzerStatisticsAsync(Document document, CancellationToken cancellationToken)
+        {
+            SyntaxTree tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+
+            SyntaxNode root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var tokensAndNodes = root.DescendantNodesAndTokensAndSelf(descendIntoTrivia: true);
+
+            int numberOfNodes = tokensAndNodes.Count(x => x.IsNode);
+            int numberOfTokens = tokensAndNodes.Count(x => x.IsToken);
+            int numberOfTrivia = root.DescendantTrivia(descendIntoTrivia: true).Count();
+
+            return new Statistic(numberOfNodes, numberOfTokens, numberOfTrivia);
+        }
+
+        private static IEnumerable<DiagnosticAnalyzer> FilterAnalyzers(IEnumerable<DiagnosticAnalyzer> analyzers, Options options)
+        {
+            foreach (var analyzer in analyzers)
+            {
+                if (options.UseAll)
+                {
+                    yield return analyzer;
+                }
+                else if (options.AnalyzerIds.Count == 0)
+                {
+                    if (analyzer.SupportedDiagnostics.Any(i => i.IsEnabledByDefault))
+                    {
+                        yield return analyzer;
+                    }
+
+                    continue;
+                }
+                else if (analyzer.SupportedDiagnostics.Any(y => options.AnalyzerIds.Contains(y.Id)))
+                {
+                    yield return analyzer;
+                }
+            }
+        }
+
+        private static ImmutableArray<Assembly> GetAnalyzerAssemblies(string path)
+        {
+            if (File.Exists(path))
+            {
+                return ImmutableArray.Create(Assembly.LoadFrom(path));
```

Comment:
if this fail due to dependency issue, what will this show to users? can we make sure if it fails due to dependencies, we show more useful information to users?

Reply:
ex) FileLoadException.FusionLog if that is there. and tell them how to make those dependencies discoverable by this tool.

Reply:
➡️ Let's deal with this as a follow-up issue (not in this PR).

Reply:
[https://github.com/dotnet/roslyn/issues/23108](https://github.com/dotnet/roslyn/issues/23108)

---
In reply to: [149861665](https://github.com/dotnet/roslyn/pull/23087#discussion_r149861665) [](ancestors = 149861665)